pkg/devicetemplateconverter/devicetemplateconverter.go

```diff
@@ -0,0 +1,81 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package devicetemplateconverter provides device template services.
+package devicetemplateconverter
+
+import (
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"go.thethings.network/lorawan-stack/pkg/component"
+	"go.thethings.network/lorawan-stack/pkg/devicetemplates"
+	"go.thethings.network/lorawan-stack/pkg/errors"
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+	"google.golang.org/grpc"
+)
+
+// Config represents the DeviceTemplateConverter configuration.
+type Config struct {
+	Enabled []string `name:"enabled" description:"Enabled converters"`
+}
+
+// DeviceTemplateConverter implements the Device Template Converter component.
+//
+// The Device Template Converter exposes the EndDeviceTemplateConverter service.
+type DeviceTemplateConverter struct {
+	*component.Component
+
+	converters map[string]devicetemplates.Converter
+
+	grpc struct {
+		endDeviceTemplateConverter *endDeviceTemplateConverterServer
+	}
+}
+
+var errNotFound = errors.DefineNotFound("converter", "converter `{id}` not found")
+
+// New returns a new *DeviceTemplateConverter.
+func New(c *component.Component, conf *Config) (*DeviceTemplateConverter, error) {
+	converters := make(map[string]devicetemplates.Converter, len(conf.Enabled))
```

Comment:
Add a context with the component field set, i.e `log.NewContextWithField(c.Context(), "namespace", "dtc")`

Reply:
https://github.com/TheThingsNetwork/lorawan-stack/pull/1016/commits/e3c02b7827de0539888162511dba908687096c9b